src/ui_utils.c

```diff
@@ -2472,46 +2472,52 @@ void ui_init_builder(void)
 }
 
 
-static void init_custom_style(void)
-{
 #if GTK_CHECK_VERSION(3, 0, 0)
-	const struct {
-		guint version;
-		const gchar *file;
-	} css_files[] = {
-		/*
-		 * Keep these from newest to oldest,
-		 * and make sure 0 remains last
-		 */
-		{ 20, "geany-3.20.css" },
-		{ 0, "geany-3.0.css" },
-	};
-	guint gtk_version = gtk_get_minor_version();
-	gsize i = 0;
-	gchar *css_file;
-	GtkCssProvider *css = gtk_css_provider_new();
+static void load_css_theme(const gchar *fn, guint priority)
+{
+	GtkCssProvider *provider = gtk_css_provider_new();
 	GError *error = NULL;
 
-	/* gtk_version will never be smaller than 0 */
-	while (css_files[i].version > gtk_version)
-		++i;
-
-	css_file = g_build_filename(app->datadir, css_files[i].file, NULL);
-	if (! gtk_css_provider_load_from_path(css, css_file, &error))
+	if (! gtk_css_provider_load_from_path(provider, fn, &error))
 	{
 		g_warning("Failed to load custom CSS: %s", error->message);
 		g_error_free(error);
+		return;
 	}
-	else
+
+	gtk_style_context_add_provider_for_screen(gdk_screen_get_default(),
+		GTK_STYLE_PROVIDER(provider), priority);
+	geany_debug("Loaded GTK+ CSS theme '%s'", fn);
+
+	g_object_unref(provider);
+}
+#endif
+
+
+// see setup_gtk2_styles() in libmain.c for GTK+ 2-specific theme initialization
+static void init_custom_style(void)
+{
+#if GTK_CHECK_VERSION(3, 0, 0)
+	gchar *theme_fn;
+
+	// load the main geany.css file from system data dir
+	theme_fn = g_build_filename(app->datadir, "geany.css", NULL);
+	load_css_theme(theme_fn, GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
+	g_free(theme_fn);
+
+	// if runtime GTK+ version is >= 3.20 then override with some needed version-specific changes
+	if (gtk_check_version(3, 20, 0) == NULL)
 	{
-		gtk_style_context_add_provider_for_screen(gdk_screen_get_default(),
-			GTK_STYLE_PROVIDER(css), GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
+		theme_fn = g_build_filename(app->datadir, "geany-3.20.css", NULL);
+		load_css_theme(theme_fn, GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
+		g_free(theme_fn);
 	}
 
-	g_object_unref(css);
-	g_free(css_file);
-#else
-	/* see setup_gtk2_styles() in main.c */
+	// if the user provided a geany.css file in their config dir, try and load that
```

Comment:
Do we really need that?  Isn't *~/.config/gtk-3.0/gtk.css* enough?  AFAIK all our styles can easily be applied to Geany only, so I don't see why we'd need extra stuff?